include/swift/Driver/DependencyGraph.h

```diff
@@ -264,6 +267,14 @@ class DependencyGraph : public DependencyGraphImpl {
   ///
   /// If you want to see how each node gets added to \p visited, pass a local
   /// MarkTracer instance to \p tracer.
+  ///
+  /// Assumes that the starting node is "cascading"; that all dependencies must
+  /// be dirty if the start is dirty. Therefore, mark the start. For each
+  /// visited node, add it to visited, and mark it if it cascades. The start
+  /// node is NOT added to visited.
```

Comment:
In this graph model, nodes aren't cascading or not; edges are. (Also, please be consistent about use of `\p visited`!)

Reply:
Good catch on "visited!" I've fixed that.